litho-widget/src/main/java/com/facebook/litho/widget/TextSpec.java

```diff
@@ -321,6 +321,12 @@ static void onMeasure(
       return;
     }
 
+    YogaDirection layoutDirection = layout.getResolvedLayoutDirection();
```

Comment:
Can we leave this code in createTextLayout and keep the signature the same? It looks like we repeat this same pre-amble when we use it. I also am not a fan that you have to know to pre-process the alignment before passing it in (whereas now you mostly just pass everything in and it figures it all out)

Reply:
The reason for pulling out the calls was to avoid performing duplicate calculations computing finalLayoutAlignment. We could address your concerns one of 3 ways: 1) keep the same signature for createTextLayout and just deal with the extra calculations, 2) Update createTextLayout with an in/out parameter for the finalLayoutAlignment to cache the result, or 3) extract those 3 lines to calculate finalLayoutAlignment into its own function but leave the rest of the code as is in the PR. Which approach would you prefer?